### PR TITLE
feat: all-product opt-in-out config

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Builds out opt-in opt-out config for all remaining products.

## 🤷 Why

To reduce boilerplate needed when onboarding a new product to HashiCorp Developer.

## 🛠️ How

- Adds new function `generateGetIoRedirectPath`, which returns a `getRedirectPath` function for a given dot-io baseUrl.
- Uses `generateGetIoRedirectPath` to set up `PLATFORM_OPTIONS` for all remaining products

## 🧪 Testing

- [ ] Test the opt-in opt-out flow with an existing product already in Beta, such as [/vault][/vault] or [/waypoint][/waypoint]
    - Not sure this is really possible to fully test without this work being in production? 🤔 Open to ideas on how to best test this work.

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1202097197789424/1202606978405428/f
[preview]: https://dev-portal-git-zsopt-in-out-config-hashicorp.vercel.app/
[/vault]: https://dev-portal-git-zsopt-in-out-config-hashicorp.vercel.app/vault
[/waypoint]: https://dev-portal-git-zsopt-in-out-config-hashicorp.vercel.app/waypoint

### CODE REVIEW

**Comment by zchsh** [comment]
@ashleemboyer @kendallstrautman @BRKalow This PR builds out opt-in-out config for remaining products. Intent here being partially to get Nomad onboarded, but partially also to reduce the need to make further changes when onboarding further products.

Let me know what ya'll think 🙇‍♂️

**Review by kendallstrautman — APPROVED** [review]
This looks good to me. I'm also unclear on the best way of testing opt in stuff for new products as we onboard. I think for .io sites, the opt in work should 'turn on' when the product is added to the beta array. [For learn ](https://github.com/hashicorp/learn/pulls?q=is%3Apr+is%3Aopen+label%3Ado-not-merge)I have a few PRs open to stage the opt in work for those products as well. But I _think_ that if someone is already opted into dev portal via learn, when we turn on that beta product on learn those redirects will automatically go into play. 

TL:DR, I think we can defer fully  testing until we are turning on those opt in banners. Lmk if that makes sense.

**Comment by BRKalow** [comment]
We should at least be able to test the opt-out flow by manually setting the relevant cookie, but yeah it's pretty tough until everything is rolled out 👍
